src/libraries/System.Private.CoreLib/src/System/Threading/CancellationTokenSource.cs

```diff
@@ -365,6 +365,53 @@ private void CancelAfter(uint millisecondsDelay)
             }
         }
 
+        /// <summary>
+        /// Attempts to reset the <see cref="CancellationTokenSource"/> to be used for an unrelated operation.
+        /// </summary>
+        /// <returns>
+        /// true if the <see cref="CancellationTokenSource"/> has not had cancellation requested and could
+        /// have its state reset to be reused for a subsequent operation; otherwise, false.
+        /// </returns>
+        /// <remarks>
+        /// <see cref="TryReset"/> is intended to be used by the sole owner of the <see cref="CancellationTokenSource"/>
+        /// when it is known that the operation with which the <see cref="CancellationTokenSource"/> was used has
+        /// completed, no one else will be attempting to cancel it (usage of <see cref="TryReset"/> concurrently with
+        /// issuing cancellation is not thread-safe), and any registrations still remaining are erroneous.  Upon a
```

Comment:
> (usage of <see cref="TryReset"/> concurrently with issuing cancellation is not thread-safe)

Turns out this is fine for kestrel, we're already doing this today.

Reply:
> Turns out this is fine for kestrel, we're already doing this today.

TryReset doesn't exist today :)  What is it Kestrel is already doing?

Reply:
Synchronizes throwing away the token and it potentially firing.